command/hcl2_upgrade.go

```diff
@@ -153,7 +153,7 @@ func (c *HCL2UpgradeCommand) RunContext(_ context.Context, cla *HCL2UpgradeArgs)
 
 	hdl, ret := c.GetConfigFromJSON(&cla.MetaArgs)
 	if ret != 0 {
-		return ret
+		c.Ui.Error(fmt.Sprintf("GetConfigFromJSON error"))
```

Comment:
This looks more like an internal error. I would change the text to be more friendly
```suggestion
		c.Ui.Error(fmt.Sprintf("Failed to get config from JSON"))
```

Reply:
Failures sound fatal.  Why not describe what is happening:

` Builder <X> not recognized. Format upgrade continuing.`

Reply:
I like the suggestion ☝🏼

Reply:
Great - Even topped-up on caffeine I wouldn't have reacted well to that error msg ;)

Reply:
`GetConfigFromJSON` will actually tell more on error, ex:

https://github.com/hashicorp/packer/blob/00e503388eb3fec2d1504bb9f82bf2ee189073ff/command/build.go#L128-L135